### ISSUE TITLE
histock: chart.series.setData(seriesData) causes stack overflow

### ISSUE THREAD

**Issue by ranair** [highcharts/highcharts#4844]
I have a highstock chart with the following configuration
chatrt1: column chart
chart2: column chart

They are sharing the same x axis and also I have a navigator.
It is based on this sample:
http://www.highcharts.com/stock/demo/candlestick-and-volume

Here is the sample code which cause this behavior:

```js
==========================================
import Ember from 'ember';
import Merger from '../services/merger';

/* globals Highcharts: true */

export default Ember.Object.extend({
    init: function () {
        this._super();
        this.set('merger', Merger.create({
            defaults: this.get('defaultChartSettings'),
            overrides: this.get('overrideChartSettings')
        }));
    },
    defaultChartSettings: {
        credits: {
            enabled: false
        },
        rangeSelector: {
            enabled: false
        },
        navigator: {
            enabled: true
        },

        title: {
            text: null
        },
        xAxis: {
            type: 'datetime',
            labels: {
                formatter: function () {
                    return Highcharts.dateFormat('%m-%d', this.value);
                }
            }
        },
        yAxis: [{
            labels: {
                align: 'right',
                x: -3
            },
            title: {
                text: null
            },
            height: '60%',
            lineWidth: 2
        }, {
            labels: {
                align: 'right',
                x: -3
            },
            title: {
                text: null
            },
            top: '65%',
            height: '35%',
            offset: 0,
            lineWidth: 2
        }],
        series: [{
            type: 'column',
            showInLegend: false,
            name: null,
            data: []
        }, {
            type: 'column',
            showInLegend: false,
            name: null,
            data: [],
            yAxis: 1
        }]
    },
    overrideChartSettings: null,
    chartSettings: Ember.computed.alias('merger.merged'),

    redraw: function (chartDrawer, chart, data) {
        // Shove the data into the chartSettings (somehow)
        chart = chartDrawer.createChart();
        this.redraw0(chartDrawer, chart, data, 'revenue', 0);
        this.redraw0(chartDrawer, chart, data, 'count', 1);
    },
    redraw0: function (chartDrawer, chart, data, dataKey, seriesIndex) {
        var seriesData = [];
        data.forEach((dataPoint) => {
            seriesData.push({x: this.getCategoryName(dataPoint), y: dataPoint.get(dataKey)});
        });
        chart.series[seriesIndex].setData(seriesData);
    }

});
==========================================
```
The error is happening when the series data is dynamically updated using setData().

But if I choose not to update the data dynamically, but set the series in the settings, then create the chart, it works...(I guess that is the default way of creating the chart)
Here is the error:

```js
==========================================
ember.debug.js:26510 RangeError: Maximum call stack size exceeded
    at Object.Series.processData (https://dashboard-apache.calliduscloud.com/dashboard_ui/assets/vendor.js:86197:41)
    at Object.seriesProto.processData (https://dashboard-apache.calliduscloud.com/dashboard_ui/assets/vendor.js:92547:22)
    at Object.seriesProto.processData (https://dashboard-apache.calliduscloud.com/dashboard_ui/assets/vendor.js:95986:20)
    at https://dashboard-apache.calliduscloud.com/dashboard_ui/assets/vendor.js:80113:12
    at Array.forEach (native)
    at each.Array.forEach.len (https://dashboard-apache.calliduscloud.com/dashboard_ui/assets/vendor.js:73663:37)
    at Object.Axis.setTickInterval (https://dashboard-apache.calliduscloud.com/dashboard_ui/assets/vendor.js:80112:4)
    at Object.Axis.setScale (https://dashboard-apache.calliduscloud.com/dashboard_ui/assets/vendor.js:80403:9)
    at Object.<anonymous> (https://dashboard-apache.calliduscloud.com/dashboard_ui/assets/vendor.js:92760:10)
    at Object.obj.(anonymous function) [as setScale] (https://dashboard-apache.calliduscloud.com/dashboard_ui/assets/vendor.js:73095:15)
```

**Comment by pawelfus** [comment]
Could you setup jsFiddle? I'm sorry, but it's nearly impossible to debug (and fix) with just part of the code. 

Anyway, when do you call `redraw()` method? It sounds like you are trying to `setData()` inside another `setData()` method.

**Comment by AO16** [comment]
I'm also having a similar issue in Ember. I will see if I can throw together a jsfiddle, unless there has been an update some where in another issue.

**Comment by TorsteinHonsi** [comment]
Okay, I will wait for the fiddle. It may be fixed now with the refactoring done for #2266.

**Comment by AO16** [comment]
The issue was fixed upon switching over to 4.2.4
